android/src/main/java/com/brentvatne/exoplayer/ExoPlayerView.kt

```diff
@@ -0,0 +1,317 @@
+            style.paddingLeft,
+            style.paddingTop,
+            style.paddingTop,
+            style.paddingBottom
+        )
+        if (style.opacity.toInt() != 0) {
+            subtitleLayout.alpha = style.opacity
+            subtitleLayout.visibility = View.VISIBLE
+        } else {
+            subtitleLayout.visibility = View.GONE
+        }
+    }
+
+    fun setShutterColor(color: Int) {
+        shutterView.setBackgroundColor(color)
+    }
+
+    fun updateSurfaceView(@ViewType.ViewType viewType: Int) {
+        this.viewType = viewType
+        var viewNeedRefresh = false
+        when (viewType) {
+            ViewType.VIEW_TYPE_SURFACE, ViewType.VIEW_TYPE_SURFACE_SECURE -> {
+                if (surfaceView !is SurfaceView) {
+                    surfaceView = SurfaceView(context)
+                    viewNeedRefresh = true
+                }
+                (surfaceView as SurfaceView).setSecure(viewType == ViewType.VIEW_TYPE_SURFACE_SECURE)
+            }
+
+            ViewType.VIEW_TYPE_TEXTURE -> {
+                if (surfaceView !is TextureView) {
+                    surfaceView = TextureView(context)
+                    viewNeedRefresh = true
+                }
+                // Support opacity properly:
+                (surfaceView as TextureView).isOpaque = false
+            }
+
+            else -> {
+                DebugLog.wtf(TAG, "Unexpected texture view type: $viewType")
+            }
+        }
+
+        if (viewNeedRefresh) {
+            surfaceView?.layoutParams = layoutParams
+
+            if (layout.getChildAt(0) != null) {
+                layout.removeViewAt(0)
+            }
+            layout.addView(surfaceView, 0, layoutParams)
+
+            if (this.player != null) {
+                setVideoView()
+            }
+        }
+    }
+
+    private fun updateShutterViewVisibility() {
+        shutterView.visibility = if (this.hideShutterView) {
+            View.INVISIBLE
+        } else {
+            View.VISIBLE
+        }
+    }
+
+    override fun requestLayout() {
+        super.requestLayout()
+        post(measureAndLayout)
+    }
+
+    // AdsLoader.AdViewProvider implementation.
+    override fun getAdViewGroup(): ViewGroup =
+        Assertions.checkNotNull(
+            adOverlayFrameLayout,
+            "exo_ad_overlay must be present for ad playback"
+        )
+
+    /**
+     * Set the {@link ExoPlayer} to use. The {@link ExoPlayer#addListener} method of the
+     * player will be called and previous
+     * assignments are overridden.
+     *
+     * @param player The {@link ExoPlayer} to use.
+     */
+    fun setPlayer(player: ExoPlayer?) {
+        if (this.player == player) {
+            return
+        }
+        if (this.player != null) {
+            this.player!!.removeListener(componentListener)
+            clearVideoView()
+        }
+        this.player = player
+        shutterView.visibility = if (this.hideShutterView) {
```

Comment:
a ternary would be easier to read here ...

Reply:
There's no ternary operator in kotlin.

https://stackoverflow.com/questions/16336500/kotlin-ternary-conditional-operator